src/io/flutter/perf/EditorPerfDecorations.java

```diff
@@ -0,0 +1,502 @@
+          if (flutterToolWindow.isVisible()) {
+            showPerfViewMessage();
+            return;
+          }
+          flutterToolWindow.show(() -> showPerfViewMessage());
+        }
+      }
+    };
+  }
+
+  private void showPerfViewMessage() {
+    final FlutterView flutterView = ServiceManager.getService(getApp().getProject(), FlutterView.class);
+    final InspectorPerfTab inspectorPerfTab = flutterView.showPerfTab(getApp());
+    final StringBuilder sb = new StringBuilder("<html><body>");
+    final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm:ss");
+    sb.append("<h2>Widget rebuild stats for ");
+    sb.append(Objects.requireNonNull(perfModelForFile.getTextEditor().getFile()).getName());
+    sb.append(" at ");
+    sb.append(formatter.format(LocalDateTime.now()));
+    sb.append("</h2>");
+    for (String line : getTooltipLines()) {
+      sb.append("<h3>");
+      sb.append(line);
+      sb.append("</h3>");
+    }
+
+    sb.append("<p>");
+    sb.append("Rebuilding widgets is generally very cheap. You should only worry " +
+              "about optimizing code to reduce the the number of widget rebuilds " +
+              "if you notice that the frame rate is bellow 60fps or if widgets " +
+              "that you did not expect to be rebuilt are rebuilt a very large " +
+              "number of times.");
+    sb.append("</p>");
+    sb.append("</body></html>");
+    inspectorPerfTab.getWidgetPerfPanel().setPerfMessage(perfModelForFile.getTextEditor(), range, sb.toString());
+  }
+
+  @NotNull
+  @Override
+  public Alignment getAlignment() {
+    return Alignment.LEFT;
+  }
+
+  @NotNull
+  @Override
+  public Icon getIcon() {
+    lastIcon = getIconInternal();
+    return lastIcon;
+  }
+
+  public Icon getIconInternal() {
+    final int count = getCountPastSecond();
+    if (count == 0) {
+      return perfModelForFile.isHoveredOverLineMarkerArea() ? FlutterIcons.CustomInfo : EMPTY_ICON;
+    }
+    if (count > HIGH_LOAD_THRESHOLD) {
+      return RED_PROGRESS;
+    }
+    return NORMAL_PROGRESS;
+  }
+
+  Color getErrorStripeMarkColor() {
+    // TODO(jacobr): tween from green or blue to red depending on the count.
+    final int count = getCountPastSecond();
+    if (count == 0) {
+      return null;
+    }
+    if (count > HIGH_LOAD_THRESHOLD) {
+      return JBColor.RED;
+    }
+    return JBColor.YELLOW; // TODO(jacobr): should we use green here instead?
+  }
+
+  public void updateUI(boolean repaint) {
+    final int count = getCountPastSecond();
+    final TextAttributes textAttributes = highlighter.getTextAttributes();
+    assert textAttributes != null;
+    boolean changed = false;
+    if (count > 0) {
+      final Color targetColor = getErrorStripeMarkColor();
+      final double animateTime = (double)(System.currentTimeMillis()) * 0.001;
+      // TODO(jacobr): consider tracking a start time for the individual
```

Comment:
Just an idea, but you could also tick an animation when a widget repaints. So the speed or rotation would be proportional to the repaint rate.

Reply:
That could be fun. The one catch is we only get updated repaint count rates once per second currently which is good for keeping the traffic on the vm service protocol low but means we can't quite show the ticks at just the right time but we could fake it base on the data we have for the previous second.

Reply:
Yeah, the position of some dial hand in a icon could give you a sense of both which specific widgets repainted, and the rate or repaint.